packages/data/README.md

```diff
@@ -281,6 +281,33 @@ const SaleButton = withDispatch( ( dispatch, ownProps ) => {
 	};
 } )( Button );
 
+// Rendered in the application:
+//
+//  <SaleButton discountPercent="20">Start Sale!</SaleButton>
+```
+
+In the majority of cases, it will be sufficient to use only two first params passed to `mapDispatchToProps` as illustrated in the previous example. However, there might be some very advanced use cases where using the `registry` object might be used as a tool to optimize the performance of your component. Using `select` function from the registry might be useful when you need to fetch some dynamic data from the store at the time when the event is fired, but at the same time, you never use it to render your component. In such scenario, you can avoid using the `withSelect` higher order component to compute such prop, which might lead to unnecessary re-renders of you component caused by its frequent value change. Keep in mind, that `mapDispatchToProps` must return an object with functions only. 
```

Comment:
I think we lost this useful information in the transition toward docgen-generated documentation, cc @nosolosw

Reply:
how so? I see the comment in the [readme](https://github.com/WordPress/gutenberg/tree/master/packages/data#withDispatch) taken from the [JSDoc block](https://github.com/WordPress/gutenberg/blob/master/packages/data/src/components/with-dispatch/index.js#L50).

Reply:
🤔 I... swear I'd seen this missing from the current documentation. But it's there, you're quite right. Sorry for falsely raising the alarms!

Reply:
No need to be sorry, I'm happily porting anything that's missing :) We ought to have great docs!